src/movies.js

```diff
@@ -1,22 +1,56 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+    const directors = moviesArray.map((movie) => movie.director);
+    return directors;
+    };
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+    return spielbergDrama.length;
+};
+
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
+    if (!moviesArray.length) return 0;
+    const summedScores = moviesArray.reduce((acc, val) => {
+        if (!val.score) return acc + 0;
+        return acc + val.score;
+    }, 0);
+    const averageScore = summedScores / moviesArray.length;
+    return Number(averageScore.toFixed(2)); 
+};
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore(moviesArray) {}
+function dramaMoviesScore(moviesArray) {
```

Comment:
In the previous iteration you already created a function that calculates the average, so you can use that function inside this new function, and in this way spare a couple of lines of code:

```
function dramaMoviesScore(moviesArray) {
  const dramaMovies = moviesArray.filter((movie) =>
    movie.genre.includes("Drama")
  );

  return scoresAverage(dramaMovies);
}
```